instance.tf

```
@@ -0,0 +1,74 @@
+resource "aws_instance" "public" {
+  ami                         = "ami-05fa00d4c63e32376"
+  associate_public_ip_address = true
+  instance_type               = "t2.micro"
+  key_name                    = "vpro-key"
+  vpc_security_group_ids      = [aws_security_group.public.id]
+  subnet_id                   = aws_subnet.public[0].id
+
+  tags = {
+    Name = "${var.env_code}-public"
+  }
+}
+
+resource "aws_security_group" "public" {
+  name        = "${var.env_code}-public"
+  description = "Allow inbound traffic"
+  vpc_id      = aws_vpc.demo-vpc.id
+
+  ingress {
+    description = "SSH from public"
+    from_port   = 22
+    to_port     = 22
+    protocol    = "tcp"
+    cidr_blocks = ["71.246.214.93/32"]
```

We can have this as a variable as well

```suggestion
    cidr_blocks = ["${var.my_public_ip}/32"]
```

But don't forget to declare the variable in variables.tf